core/app/[locale]/(default)/(auth)/register/_actions/register-customer.ts

```diff
@@ -34,72 +39,156 @@ const RegisterCustomerMutation = graphql(`
   }
 `);
 
-type Variables = VariablesOf<typeof RegisterCustomerMutation>;
-type RegisterCustomerInput = Variables['input'];
-
-interface RegisterCustomerForm {
-  formData: FormData;
-  reCaptchaToken?: string;
+function parseRegisterCustomerInput(
+  value: Record<string, string | number | string[] | undefined>,
+  fields: Array<Field | FieldGroup<Field>>,
+): VariablesOf<typeof RegisterCustomerMutation>['input'] {
+  const customFields = fields
+    .flatMap((f) => (Array.isArray(f) ? f : [f]))
+    .filter(
+      (field) =>
+        ![
+          String(FieldNameToFieldId.email),
+          String(FieldNameToFieldId.password),
+          String(FieldNameToFieldId.confirmPassword),
+          String(FieldNameToFieldId.firstName),
+          String(FieldNameToFieldId.lastName),
+          String(FieldNameToFieldId.phone),
+          String(FieldNameToFieldId.company),
+        ].includes(field.name),
+    );
+
+  return {
+    email: String(value[FieldNameToFieldId.email] ?? ''),
+    password: String(value[FieldNameToFieldId.password] ?? ''),
+    firstName: String(value[FieldNameToFieldId.firstName] ?? ''),
+    lastName: String(value[FieldNameToFieldId.lastName] ?? ''),
+    phone: String(value[FieldNameToFieldId.phone] ?? ''),
+    company: String(value[FieldNameToFieldId.company] ?? ''),
```

Comment:
Is it appropriate for these to be empty strings instead of `null` or something else? I'm not familiar enough with the BigCommerce GraphQL API to know (same applies to other instances of this pattern below)

Reply:
Refactored to parsing with zod.